velox/functions/sparksql/RaiseError.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct RaiseErrorFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void callNullable(
+      out_type<UnknownValue>& result,
+      const arg_type<Varchar>* input) {
+    if (input) {
+      VELOX_USER_FAIL(*input);
+    }
+    VELOX_USER_FAIL("");
```

Comment:
We can follow below implementation to return status.
https://github.com/facebookincubator/velox/commit/194772c823e73f3847ab1754a70bcda55a6c0679#diff-2a2739960fe78b23b6ac90af85cef58647341837a9cfec78d3d66eb90ea4b1b0R29-R36

Reply:
actually i tried, but based on my local test, return status way work for `call`, however if change to return status way for 
``callNullable`, it would causes compile issue.

for example below change 

```

template <typename T>
struct RaiseErrorFunction {
  VELOX_DEFINE_FUNCTION_TYPES(T);

  FOLLY_ALWAYS_INLINE Status callNullable(
      out_type<UnknownValue>& result,
      const arg_type<Varchar>* input) {
    return Status::UserError("");
  }
};
```
would cause
```
xxx/velox/functions/Registerer.h:74:51:   required from ‘bool facebook::velox::registerFunction(const std::vector<std::__cxx11::basic_string<char> >&, const std::vector<facebook::velox::exec::SignatureVariable>&, bool) [with Func = facebook::velox::functions::sparksql::RaiseErrorFunction; TReturn = facebook::velox::UnknownValue; TArgs = {facebook::velox::Varchar}]’
/home/gayangya/Work/velox/velox/functions/sparksql/Register.cpp:465:62:   required from here
/home/gayangya/Work/velox/./velox/core/SimpleFunctionMetadata.h:804:44: error: static assertion failed: UDF must implement at least one of `call`, `callNullable`, or `callNullFree` functions.
This error happens also if the output and input types of the functions do not match the
ones used in registration.
  804 |       udf_has_call || udf_has_callNullable || udf_has_callNullFree,
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/home/gayangya/Work/velox/./velox/core/SimpleFunctionMetadata.h:804:44: note: ‘((((bool)facebook::velox::core::UDFHolder<facebook::velox::functions::sparksql::RaiseErrorFunction<facebook::velox::exec::VectorExec>, facebook::velox::exec::VectorExec, facebook::velox::UnknownValue, facebook::velox::ConstantChecker<facebook::velox::Varchar
```

Reply:
I see. It looks like callNullable is not extended to return status. cc: @mbasmanova 
https://github.com/facebookincubator/velox/blob/main/velox/core/SimpleFunctionMetadata.h#L734-L736

Reply:
Seems a bug. Maybe, we should fix it firstly.

Reply:
kindly ping @rui-mo

Reply:
Just created one issue: https://github.com/facebookincubator/velox/issues/10265. I will fix it, assuming we need to do that.